### ISSUE TITLE
Alerting integrations

### ISSUE THREAD

**Issue by Samrose-Ahmed** [matanolabs/matano#18]
Tracking for integrating Matano alerts with external systems

### Destinations
(prioritize)
* [ ] Pagerduty
* [ ] Slack
* [ ] -

#### Integration
@shaeqahmed 

* Sending rule matches, alerts, schema
* Updating alerts

**Comment by gdrapp** [comment]
Hey guys, great to see this on the roadmap. Would love to see [TheHive](https://thehive-project.org/)  on the list as well.

**Comment by rams3sh** [comment]
Hey Guys , 

It would be really nice to have below as part of external integrations in general.

1.Webhook -  This can solve lot of generic use cases.
2. Step function  - This will help in triggering a step function based playbook where one can have their own response playbook defined for a given alert.

**Comment by Samrose-Ahmed** [comment]
Great suggestions, especially step functions. Adding to the list.

**Comment by rams3sh** [comment]
Yeah , I have used socless and this idea comes from there . Stepfunctions also helps in approval based workflow where a user can be reached out to confirm if a given alert is indeed triggered due to a genuine use or any unauthorised use which can further be taken ahead by SOC Analyst.  This reduces burden on SOC analyst to address every alert. 

Adding the link to socless framework to provide context to the ask. 
Doc Link: https://twilio-labs.github.io/socless/
Reference to human based confirmation in socless : https://twilio-labs.github.io/socless/tutorial-interacting-with-humans-via-slack/

**Comment by grue** [comment]
Are you looking for contributions on this front, or is this something actively in progress?

**Comment by Samrose-Ahmed** [comment]
I believe @shaeqahmed was working on some integrations, starting with a Slack integration, but we would be happy to accept any contributions. Is there a specific integration you are interested in?

**Comment by rams3sh** [comment]
It would be nice to have standalone AWS lambda as a trigger as well in case there is not much complex workflow to have a step function.

**Comment by timoguin** [comment]
> It would be nice to have standalone AWS lambda as a trigger as well in case there is not much complex workflow to have a step function.

SNS was recently added as a destination. You could hook SQS + Lambda into that topic.
